aws/resource_aws_acm_certificate.go

```diff
@@ -137,19 +203,21 @@ func resourceAwsAcmCertificateRead(d *schema.ResourceData, meta interface{}) err
 			return resource.NonRetryableError(err)
 		}
 
-		domainValidationOptions, emailValidationOptions, err := convertValidationOptions(resp.Certificate)
+		certificateDetails, err := convertCertificateDetails(resp.Certificate)
+
+		if len(certificateDetails) < 1 {
+			return resource.NonRetryableError(fmt.Errorf("Error getting certificate details"))
+		}
 
 		if err != nil {
 			return resource.RetryableError(err)
 		}
 
-		if err := d.Set("domain_validation_options", domainValidationOptions); err != nil {
```

Comment:
If we are going to deprecate the old attributes, they still need to be set in the state to not break backwards compatibility.

Reply:
I spent the morning trying to figure out how to not break backwards compatibility and I just couldn't do it. The problem I am having is that the old attribute has an unfortunate name collision with the new attribute and the attributers are fundamentally different types. The old attribute was a computed output, the new one is an optional input. So when I set the deprecated attributes for an optional input like so:
```
		//support for deprecated attributes
		d.Set("validation_emails", certificateDetails[0]["validation_emails"])
		domainValidationOptionsInput := d.Get("domain_validation_options")
		for _, o := range domainValidationOptionsInput.([]interface{}) {
			x := o.(map[string]interface{})
			x["resource_record_name"] = certificateDetails[0]["resource_record_name"]
			x["resource_record_type"] = certificateDetails[0]["resource_record_type"]
			x["resource_record_value"] = certificateDetails[0]["resource_record_value"]
		}
		d.Set("domain_validation_options", domainValidationOptionsInput)
```
then when I try to create the resource using the deprecated approach, i get the following error: 
```
Resource 'aws_acm_certificate.cert' does not have attribute 'domain_validation_options.0.resource_record_name' for variable 'aws_acm_certificate.cert.domain_validation_options.0.resource_record_name'
```
This isn't a problem with validation_emails, because there is no name collision for that attribute, just domain_validation_options. Do you have any suggestions?

Also, as for running the acceptance tests --- it takes a really long time and I really can't afford the time or cost to run the full suite of tests. Is there a way to just test the couple acm test files I want to test? When I do a `go test /aws/resource_aws_acm_certificate_test.go` i get an error about some functions not being recognized. I am not sure which test files are dependen cies for my test file to run.